nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -206,22 +210,59 @@ void
 BtNavigator::onLoop()
 {
   // action server feedback (pose, duration of task,
-  // number of recoveries, and distance remaining to goal)
+  // time remaining, number of recoveries, and
+  // distance remaining to goal)
   auto feedback_msg = std::make_shared<Action::Feedback>();
 
+  geometry_msgs::msg::PoseStamped current_pose;
   nav2_util::getCurrentPose(
-    feedback_msg->current_pose, *tf_, global_frame_, robot_frame_, transform_tolerance_);
+    current_pose, *tf_, global_frame_, robot_frame_, transform_tolerance_);
 
   auto blackboard = bt_action_server_->getBlackboard();
 
-  geometry_msgs::msg::PoseStamped goal_pose;
-  blackboard->get(goal_blackboard_id_, goal_pose);
-
-  feedback_msg->distance_remaining = nav2_util::geometry_utils::euclidean_distance(
-    feedback_msg->current_pose.pose, goal_pose.pose);
+  // Get current path points
+  nav_msgs::msg::Path current_path;
+  blackboard->get("path", current_path);
```

Comment:
I think you didn't set the template get for path, so maybe couldn't imply the cast. Also I don't see path in the https://github.com/ros-planning/navigation2/blob/3b36c29989b429a60c5e8161aa4cd57637b81707/nav2_behavior_tree/include/nav2_behavior_tree/bt_conversions.hpp string conversion list